### DISCUSSION TITLE
version 4.3.0

### DISCUSSION THREAD

**Discussion by hvianna** [hvianna/audioMotion-analyzer#59] [Announcements]
### Added:

+ **'dual-horizontal'** option for [`channelLayout`](README.md#channellayout-string);

### Fixed:

+ Fixed opacity for [reflex effect](README.md#reflexratio-number) on *dual-combined* channel layout.

### Improved:

+ Improved [`maxFPS`](README.md#maxfps-number) frame rate throttling.

---
Release: [version 4.3.0](https://github.com/hvianna/audioMotion-analyzer/releases/tag/4.3.0)

**Comment by TF3RDL** [comment]
Awesome update, I like how `dual-horizontal` layout on your spectrum analyzer is the closest thing to Channel Spectrum panel for Default UI and 64-bit foobar2000 despite not being a foobar2000 component

Also, adding the mode where frequency labels follows decades [like this one](https://codepen.io/TF3RDL/pen/poQJwRW) would be cool since the Enhanced Spectrum analyzer component that I have on my foobar2000 x64 has this frequency lines following decades

**Comment by TF3RDL** [comment]
A Mid/Side mode like [this](https://codepen.io/TF3RDL/pen/VwENJbB) for dual-channel visualization would be cool to see, but here's a catch; AnalyserNode's built-in FFT outputs magnitude values in decibels rather than complex-valued data so doing M/S encode on frequency-domain data won't be proper M/S (unless you're using custom FFT method, which usually output real/imaginary values and you can do complex addition and subtraction on these before being converted into magnitude values to get a proper M/S spectrum visualization from frequency-domain data)

As for the M/S spectrum using the from the aforementioned CodePen project, it can be easily done on L/R data gathered by getFloatTimeDomainData on left and right channels after ChannelSplitterNode to do M/S encode before FFT (the first method does involve only simple ChannelSplitterNode/ChannelMergerNode setup to visualize separate channels but it is only doable on custom FFT implementation), the second method is use complex ChannelSplitterNode/ChannelMergerNode (to split up the channels and merge it back after M/S encode) and GainNode (to phase-invert right channel part of side channel on M/S representation) setup to do M/S visualization without having to use custom FFT method
